code/web/src/modules/crate/Item.js

```diff
@@ -1,92 +1,111 @@
 // Imports
-import React, { PureComponent } from 'react'
-import PropTypes from 'prop-types'
-import { connect } from 'react-redux'
-import { Link, withRouter } from 'react-router-dom'
+import React, { PureComponent } from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { Link, withRouter } from "react-router-dom";
 
 // UI Imports
-import Card from '../../ui/card/Card'
-import Button from '../../ui/button/Button'
-import H4 from '../../ui/typography/H4'
-import Icon from '../../ui/icon'
-import { white, grey2, black } from '../../ui/common/colors'
+import Card from "../../ui/card/Card";
+import Button from "../../ui/button/Button";
+import H4 from "../../ui/typography/H4";
+import Icon from "../../ui/icon";
+import { white, grey2, black } from "../../ui/common/colors";
 
 // App Imports
-import { APP_URL } from '../../setup/config/env'
-import userRoutes from '../../setup/routes/user'
-import { messageShow, messageHide } from '../common/api/actions'
-import { create } from '../subscription/api/actions'
+import { APP_URL } from "../../setup/config/env";
+import userRoutes from "../../setup/routes/user";
+import { messageShow, messageHide } from "../common/api/actions";
+import { create } from "../subscription/api/actions";
 
 // Component
 class Item extends PureComponent {
-
   constructor(props) {
-    super(props)
+    super(props);
 
     this.state = {
-      isLoading: false
-    }
+      isLoading: false,
+    };
   }
 
   onClickSubscribe = (crateId) => {
     this.setState({
-      isLoading: true
-    })
+      isLoading: true,
+    });
 
-    this.props.messageShow('Subscribing, please wait...')
+    this.props.messageShow("Subscribing, please wait...");
 
-    this.props.create({ crateId })
-      .then(response => {
+    this.props
+      .create({ crateId })
+      .then((response) => {
         if (response.data.errors && response.data.errors.length > 0) {
-          this.props.messageShow(response.data.errors[0].message)
+          this.props.messageShow(response.data.errors[0].message);
+        } else if (!this.props.user.details.stylePreference) {
+          this.props.messageShow("Please take the style survey!");
+          this.props.history.push(userRoutes.stylePreference.path);
         } else {
-          this.props.messageShow('Subscribed successfully.')
-
-          this.props.history.push(userRoutes.subscriptions.path)
+          this.props.messageShow("Subscribed successfully.");
+          this.props.history.push(userRoutes.subscriptions.path);
         }
       })
-      .catch(error => {
-        this.props.messageShow('There was some error subscribing to this crate. Please try again.')
+      .catch((error) => {
+        this.props.messageShow(
+          "There was some error subscribing to this crate. Please try again."
```

Comment:
Noticed a typo here

Reply:
Seen. We will address this on the next go round…